libraryBackend/libraryApp/models.py

```diff
@@ -156,22 +156,32 @@ def __str__(self):
 
 
 class Book(models.Model):
-    isbn = models.CharField(max_length=13)
+    isbn = models.CharField(max_length=13, unique=True)
     name = models.CharField(max_length=30)
     publisher = models.CharField(max_length=30)
     author = models.CharField(max_length=30)
     publish_year = models.CharField(max_length=4)
-    quantity = models.IntegerField()
     demand = models.BigIntegerField()
-    is_available = models.BooleanField(default=True)
 
     def __str__(self):
-        return self.name + "_" + self.author
+        return self.name + "_" + self.author + "_" + str(self.id)
+
+    @property
+    def quantity(self):
+        return self.copies.objects.filter(is_available=True).count()
 
 
 class Copy(models.Model):
+    barcode = models.CharField(max_length=30, unique=True)
     book = models.ForeignKey(Book, on_delete=models.CASCADE)
-    condition = models.CharField(max_length=30)
+
+    CONDITION_CHOICES = (
+        ("BEST", "BEST"),
+        ("GOOD", "GOOD"),
+        ("WORST", "WORST"),
+    )
+    condition = models.CharField(choices=CONDITION_CHOICES, max_length=30)
+    is_available = models.BooleanField(default=True)
```

Comment:
This can be checked using a property which filters over issued book and/or waiting list